neuralforecast/models/timellm.py

```diff
@@ -0,0 +1,497 @@
+# AUTOGENERATED! DO NOT EDIT! File to edit: ../../nbs/models.timellm.ipynb.
+
+# %% auto 0
+__all__ = ['ReplicationPad1d', 'TokenEmbedding', 'PatchEmbedding', 'FlattenHead', 'ReprogrammingLayer', 'Normalize', 'TimeLLM']
+
+# %% ../../nbs/models.timellm.ipynb 6
+import math
+import numpy as np
+from typing import Optional
+
+import torch
+import torch.nn as nn
+
+from ..common._base_windows import BaseWindows
+
+from ..losses.pytorch import MAE
+
+from transformers import GPT2Config, GPT2Model, GPT2Tokenizer
```

Comment:
since `transfomers` is not a core dependency, we can try to see if it is installed and then throw an error when it needs to be used for default arguments (`llm=None`), something like:

```python
IS_TRANSFORMERS_INSTALLED = True
try:
     from transformers import GPT2Config, GPT2Model, GPT2Tokenizer
except ImportError:
     IS_TRANSFORMERS_INSTALLED = False